src/Workspaces/Core/Desktop/Workspace/Host/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -30,8 +31,58 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
         /// </summary>
         internal class TemporaryStorageService : ITemporaryStorageService2
         {
+            /// <summary>
+            /// The maximum size of a single storage unit in a memory mapped file which is shared with other storage
+            /// units.
+            /// </summary>
+            private const long SingleFileThreshold = 128 * 1024;
+
+            /// <summary>
+            /// The size of a memory mapped file created to store multiple temporary objects.
+            /// </summary>
+            private const long MultiFileBlockSize = SingleFileThreshold * 32;
```

Comment:
If a multifile block is around 4MB, then why would we not allow it to store files that were, say, 256k in it?